esptool.py

```diff
@@ -653,7 +668,8 @@ def set_data_lengths(mosi_bits, miso_bits):
         if data_bits == 0:
             self.write_reg(SPI_W0_REG, 0)  # clear data register before we read it
         else:
-            data = pad_to(data, 4, b'\00')  # pad to 32-bit multiple
```

Comment:
I see the problem now - esp-idf is using an older version (e9e917) so I didn't see that master here has something newer. Will get it straight with the rest of the fixes here.